### PR TITLE
implement some function！

### PR DESCRIPTION
**Pull request by JamePeng**
implement function ListDataTitleTicketInfos()
implement GetOutputFormat and GetInputFormat
Implement function ReplySleepQuery
Correct and Update SRV code: implement function Subscribe

### CODE REVIEW

**Comment by neobrain** [comment]
Hi, this is neobot, using neobrain's account. I'm keeping an archive of versions of this PR:

2016-02-10T19:21:27Z: https://github.com/citra-emu/citra/compare/f1d1049c4fdf51fd382f9afb6b5cd5f104e71b0e...JamePeng:c6198deec8fa57677d1e8218fbd336ff4cda07c1

**Comment by neobrain** [comment]
Hi, this is neobot, using neobrain's account. I'm keeping an archive of versions of this PR:

2016-02-10T19:26:51Z: https://github.com/citra-emu/citra/compare/f1d1049c4fdf51fd382f9afb6b5cd5f104e71b0e...JamePeng:1fea593c7e2004dbec45dfdfe1616d49679e2641

File: src/core/hle/service/am/am.cpp [hunk 1]
```diff
@@ -44,6 +44,20 @@ void GetNumContentInfos(Service::Interface* self) {
     LOG_WARNING(Service_AM, "(STUBBED) called");
 }
 
+void ListDataTitleTicketInfos(Service::Interface* self) {
+    u32* cmd_buff = Kernel::GetCommandBuffer();
+    u32 ticket_count = cmd_buff[1];
+    u64 title_id = cmd_buff[2];
```
Comment (Subv) [inline src/core/hle/service/am/am.cpp]: u64 title_id = ((u64)cmd_buff[2] << 32) | cmd_buff[3];

File: src/core/hle/service/am/am.cpp [hunk 2]
```diff
@@ -44,6 +44,20 @@ void GetNumContentInfos(Service::Interface* self) {
...
+    title_id = (title_id << 32) | cmd_buff[3];
+    u32 start_index = cmd_buff[4];
+    u32 tipointer = cmd_buff[6];
+
+    cmd_buff[1] = RESULT_SUCCESS.raw;
+    cmd_buff[2] = 0x18;
+    LOG_WARNING(Service_AM, "(STUBBED) TicketCount=0x%08X TitleID==0x%16X StartIndex=0x%08X",
```
Comment (Subv) [inline src/core/hle/service/am/am.cpp]: %16X will not work as you expect. Use a 64 bit format specifier

File: src/core/hle/service/am/am.cpp [hunk 3]
```diff
@@ -44,6 +44,20 @@ void GetNumContentInfos(Service::Interface* self) {
...
+void ListDataTitleTicketInfos(Service::Interface* self) {
+    u32* cmd_buff = Kernel::GetCommandBuffer();
+    u32 ticket_count = cmd_buff[1];
+    u64 title_id = cmd_buff[2];
+    title_id = (title_id << 32) | cmd_buff[3];
+    u32 start_index = cmd_buff[4];
+    u32 tipointer = cmd_buff[6];
+
+    cmd_buff[1] = RESULT_SUCCESS.raw;
+    cmd_buff[2] = 0x18;
```
Comment (Subv) [inline src/core/hle/service/am/am.cpp]: What does this value mean?

Comment (JamePeng) [inline src/core/hle/service/am/am.cpp]: @Subv 
http://3dbrew.org/wiki/AMPXI:ListDataTitleTicketInfos

File: src/core/hle/service/y2r_u.cpp [hunk 4]
```diff
@@ -84,13 +84,22 @@ ResultCode ConversionConfiguration::SetStandardCoefficient(StandardCoefficient s
...
     conversion.input_format = static_cast<InputFormat>(cmd_buff[1]);
     LOG_DEBUG(Service_Y2R, "called input_format=%hhu", conversion.input_format);
 
     cmd_buff[1] = RESULT_SUCCESS.raw;
 }
 
+static void GetInputFormat(Service::Interface* self) {
+    u32* cmd_buff = Kernel::GetCommandBuffer();
+
+    InputFormat inf = conversion.input_format;
```
Comment (lioncash) [inline src/core/hle/service/y2r_u.cpp]: `inf` -> `input_format`

File: src/core/hle/service/y2r_u.cpp [hunk 5]
```diff
@@ -100,6 +109,16 @@ static void SetOutputFormat(Service::Interface* self) {
     cmd_buff[1] = RESULT_SUCCESS.raw;
 }
 
+static void GetOutputFormat(Service::Interface* self) {
+    u32* cmd_buff = Kernel::GetCommandBuffer();
+
+    OutputFormat opf = conversion.output_format;
```
Comment (lioncash) [inline src/core/hle/service/y2r_u.cpp]: `opf` -> `output_format`

File: src/core/hle/service/y2r_u.cpp (hunk 4 above)
Comment (mattvail) [inline src/core/hle/service/y2r_u.cpp]: You don't need to declare this locally

File: src/core/hle/service/y2r_u.cpp (hunk 5 above)
Comment (mattvail) [inline src/core/hle/service/y2r_u.cpp]: Ditto on declaring locally

File: src/core/hle/service/y2r_u.cpp (hunk 4 above)
Comment (lioncash) [inline src/core/hle/service/y2r_u.cpp]: Ah, true enough

**Comment by mattvail** [comment]
Your commit message only mentions ListDataTitleTicketInfos, but you implement GetInputFormat and GetOutputFormat. Since they're not in the same module, it'd be nice if it was menionned :)

File: src/core/hle/service/am/am.cpp [hunk 6]
```diff
@@ -44,6 +44,20 @@ void GetNumContentInfos(Service::Interface* self) {
     LOG_WARNING(Service_AM, "(STUBBED) called");
 }
 
+void ListDataTitleTicketInfos(Service::Interface* self) {
+    u32* cmd_buff = Kernel::GetCommandBuffer();
+    u32 ticket_count = cmd_buff[1];
+    u64 title_id = cmd_buff[2];
+    title_id = (title_id << 32) | cmd_buff[3];
+    u32 start_index = cmd_buff[4];
+    u32 tipointer = cmd_buff[6];
```
Comment (lioncash) [inline src/core/hle/service/am/am.cpp]: ticket_info_pointer

**Comment by neobrain** [comment]
Hi, this is neobot, using neobrain's account. I'm keeping an archive of versions of this PR:

2016-02-10T19:48:22Z: https://github.com/citra-emu/citra/compare/f1d1049c4fdf51fd382f9afb6b5cd5f104e71b0e...JamePeng:6448d788c56ed22acb1c175b8292be578e1eb850

**Comment by JamePeng** [comment]
@mattvail Ha :) Thank for you mention!

File: src/core/hle/service/am/am.cpp [hunk 7]
```diff
@@ -44,6 +44,19 @@ void GetNumContentInfos(Service::Interface* self) {
...
+void ListDataTitleTicketInfos(Service::Interface* self) {
+    u32* cmd_buff = Kernel::GetCommandBuffer();
+    u32 ticket_count = cmd_buff[1];
+    u64 title_id = (title_id << 32) | cmd_buff[3];
+    u32 start_index = cmd_buff[4];
+    u32 tipointer = cmd_buff[6];
+
+    cmd_buff[1] = RESULT_SUCCESS.raw;
+    cmd_buff[2] = 0x18;
+    LOG_WARNING(Service_AM, "(STUBBED) TicketCount=0x%08X TitleID==0x%I64x StartIndex=0x%08X",
```
Comment (lioncash) [inline src/core/hle/service/am/am.cpp]: The correct macro here is PRIx64

File: src/core/hle/service/am/am.cpp [hunk 8]
```diff
@@ -44,6 +44,19 @@ void GetNumContentInfos(Service::Interface* self) {
     LOG_WARNING(Service_AM, "(STUBBED) called");
 }
 
+void ListDataTitleTicketInfos(Service::Interface* self) {
+    u32* cmd_buff = Kernel::GetCommandBuffer();
+    u32 ticket_count = cmd_buff[1];
+    u64 title_id = (title_id << 32) | cmd_buff[3];
```
Comment (lioncash) [inline src/core/hle/service/am/am.cpp]: You need a cast here, and the shifted variable is incorrect.

```cpp
u64 title_id = static_cast<u64>(cmd_buff[2]) << 32 | cmd_buff[3];
```

Currently, this code has undefined behavior.

File: src/core/hle/service/am/am.cpp [hunk 9]
```diff
@@ -44,6 +44,19 @@ void GetNumContentInfos(Service::Interface* self) {
...
+    u32 start_index = cmd_buff[4];
+    u32 tipointer = cmd_buff[6];
+
+    cmd_buff[1] = RESULT_SUCCESS.raw;
+    cmd_buff[2] = 0x18;
+    LOG_WARNING(Service_AM, "(STUBBED) TicketCount=0x%08X TitleID==0x%I64x StartIndex=0x%08X",
+                "TicketInfosPointer=0x%08X",ticket_count, title_id, start_index, tipointer);
```
Comment (lioncash) [inline src/core/hle/service/am/am.cpp]: space after the comma

**Comment by neobrain** [comment]
Hi, this is neobot, using neobrain's account. I'm keeping an archive of versions of this PR:

2016-02-10T20:29:07Z: https://github.com/citra-emu/citra/compare/f1d1049c4fdf51fd382f9afb6b5cd5f104e71b0e...JamePeng:86e6d94f56c4a65cbd89347fb332c282f0987884

**Comment by neobrain** [comment]
Hi, this is neobot, using neobrain's account. I'm keeping an archive of versions of this PR:

2016-02-10T20:35:38Z: https://github.com/citra-emu/citra/compare/f1d1049c4fdf51fd382f9afb6b5cd5f104e71b0e...JamePeng:60f8b5a535e98f0eae7bb34162995b00fd224bfc

File: src/core/hle/service/y2r_u.cpp [hunk 10]
```diff
@@ -84,13 +84,21 @@ ResultCode ConversionConfiguration::SetStandardCoefficient(StandardCoefficient s
 
 static void SetInputFormat(Service::Interface* self) {
     u32* cmd_buff = Kernel::GetCommandBuffer();
-
```
Comment (lioncash) [inline src/core/hle/service/y2r_u.cpp]: There's no need to remove this line.

File: src/core/hle/service/am/am.cpp [hunk 11]
```diff
@@ -1,6 +1,7 @@
 // Copyright 2015 Citra Emulator Project
 // Licensed under GPLv2 or any later version
 // Refer to the license.txt file included.
+#include <cinttypes>
```
Comment (lioncash) [inline src/core/hle/service/am/am.cpp]: There should be a new line after the license header before includes.

File: src/core/hle/service/am/am.cpp [hunk 12]
```diff
@@ -44,6 +45,18 @@ void GetNumContentInfos(Service::Interface* self) {
...
+void ListDataTitleTicketInfos(Service::Interface* self) {
+    u32* cmd_buff = Kernel::GetCommandBuffer();
+    u32 ticket_count = cmd_buff[1];
+    u64 title_id = static_cast<u64>(cmd_buff[2]) << 32 | cmd_buff[3];
+    u32 start_index = cmd_buff[4];
+    u32 tipointer = cmd_buff[6];
+
+    cmd_buff[1] = RESULT_SUCCESS.raw;
+    cmd_buff[2] = 0x18;
+    LOG_WARNING(Service_AM, "(STUBBED) TicketCount=0x%08X TitleID=0x%016" PRIx64 " StartIndex=0x%08X TicketInfosPointer=0x%08X",ticket_count, title_id, start_index, tipointer);
```
Comment (lioncash) [inline src/core/hle/service/am/am.cpp]: There should be a space after the comma for `ticket_count`

File: src/core/hle/service/am/am.cpp [hunk 13]
```diff
@@ -44,6 +45,18 @@ void GetNumContentInfos(Service::Interface* self) {
     LOG_WARNING(Service_AM, "(STUBBED) called");
 }
 
+void ListDataTitleTicketInfos(Service::Interface* self) {
+    u32* cmd_buff = Kernel::GetCommandBuffer();
+    u32 ticket_count = cmd_buff[1];
+    u64 title_id = static_cast<u64>(cmd_buff[2]) << 32 | cmd_buff[3];
+    u32 start_index = cmd_buff[4];
+    u32 tipointer = cmd_buff[6];
```
Comment (lioncash) [inline src/core/hle/service/am/am.cpp]: `title_info_pointer`

**Comment by neobrain** [comment]
Hi, this is neobot, using neobrain's account. I'm keeping an archive of versions of this PR:

2016-02-10T21:13:19Z: https://github.com/citra-emu/citra/compare/f1d1049c4fdf51fd382f9afb6b5cd5f104e71b0e...JamePeng:80a82acf5587b395dd29fe7b1e9eed94e16c70be

**Comment by neobrain** [comment]
Hi, this is neobot, using neobrain's account. I'm keeping an archive of versions of this PR:

2016-02-10T21:20:59Z: https://github.com/citra-emu/citra/compare/f1d1049c4fdf51fd382f9afb6b5cd5f104e71b0e...JamePeng:8adfde4443e357680a022b9cd2d001885dc00bb7

File: src/core/hle/service/y2r_u.cpp [hunk 14]
```diff
@@ -91,6 +91,16 @@ static void SetInputFormat(Service::Interface* self) {
     cmd_buff[1] = RESULT_SUCCESS.raw;
 }
 
+static void GetInputFormat(Service::Interface* self) {
+    u32* cmd_buff = Kernel::GetCommandBuffer();
+
+    InputFormat input_format = conversion.input_format;
```
Comment (lioncash) [inline src/core/hle/service/y2r_u.cpp]: Why were the variables reintroduced? Like @mattvail had previously said, there's no need to do this.

**Comment by neobrain** [comment]
Hi, this is neobot, using neobrain's account. I'm keeping an archive of versions of this PR:

2016-02-10T21:54:54Z: https://github.com/citra-emu/citra/compare/f1d1049c4fdf51fd382f9afb6b5cd5f104e71b0e...JamePeng:3994c85070286891383656a2f250c77b9b549b64

**Comment by neobrain** [comment]
Hi, this is neobot, using neobrain's account. I'm keeping an archive of versions of this PR:

2016-02-11T18:41:40Z: https://github.com/citra-emu/citra/compare/f1d1049c4fdf51fd382f9afb6b5cd5f104e71b0e...JamePeng:705e074e59dfbce34b8eb3205834c49ae3b2c796

**Comment by neobrain** [comment]
Hi, this is neobot, using neobrain's account. I'm keeping an archive of versions of this PR:

2016-02-11T19:18:30Z: https://github.com/citra-emu/citra/compare/f1d1049c4fdf51fd382f9afb6b5cd5f104e71b0e...JamePeng:2a0862ad8538ec3e5a99d200c7729ee560faf8ff

**Comment by JamePeng** [comment]
@lioncash So what else can I do with this version request? Merge?

File: src/core/hle/service/apt/apt.h [hunk 15]
```diff
@@ -31,6 +31,13 @@ struct AppletStartupParameter {
     u8* data = nullptr;
 };
 
+enum class QueryReply : u32
+{
+    REPLY_REJECT        = 0x0,
+    REPLY_ACCEPT        = 0x1,
+    REPLY_LATER         = 0x2
```
Comment (lioncash) [inline src/core/hle/service/apt/apt.h]:
```cpp
enum class QueryReply : u32
{
    Reject,
    Accept,
    Later
};
```

File: src/core/hle/service/apt/apt.h [hunk 16]
```diff
@@ -89,6 +106,17 @@ void Initialize(Service::Interface* self);
...
 
 /**
+* APT::ReplySleepQuery service function
+*  Inputs:
+*      1 : AppID
+*      2 : ReplySleepQuery
+*  Outputs:
+*      1 : Result of function, 0 on success, otherwise error code
+*/
+
```
Comment (lioncash) [inline src/core/hle/service/apt/apt.h]: unnecessary newline

File: src/core/hle/service/apt/apt.h [hunk 17]
```diff
@@ -258,16 +286,17 @@ void PrepareToStartApplication(Service::Interface* self);
...
+*  Inputs:
+*      1 : Utility
+*      2 : Input Size
+*      3 : Output Size
+*      4:  Translation descriptor: (Input Size << 14) | 0x402
+*      5 : void*, Input
+*  Outputs:
+*      1 : Result of function, 0 on success, otherwise error code
+*      2 : Applet Result
+*/
```
Comment (lioncash) [inline src/core/hle/service/apt/apt.h]: `*` should be aligned

**Comment by neobrain** [comment]
Hi, this is neobot, using neobrain's account. I'm keeping an archive of versions of this PR:

2016-02-12T14:23:55Z: https://github.com/citra-emu/citra/compare/f1d1049c4fdf51fd382f9afb6b5cd5f104e71b0e...JamePeng:80c5180d4f25d1792a73d684f6f2bf67b224ec05

**Comment by JamePeng** [comment]
Anything else?

**Comment by lioncash** [comment]
squash your changes (see [here](https://gist.github.com/lioncash/fbf57f875575d3e13484) for a basic guide I made on how to do that, or look at git's documentation)

**Comment by JamePeng** [comment]
@lioncash OK ,I finish the operation, this is a nice tutorial

**Comment by lioncash** [comment]
You need to push your changes with `git push -f <your remote here> <your branch name here>`

**Comment by JamePeng** [comment]
What the hell? CMAKE something wrong!

**Comment by neobrain** [comment]
Hi, this is neobot, using neobrain's account. I'm keeping an archive of versions of this PR:

2016-02-12T17:46:23Z: https://github.com/citra-emu/citra/compare/f1d1049c4fdf51fd382f9afb6b5cd5f104e71b0e...JamePeng:724e0fe42ad6265a40b568559bd40cfafc708246

**Comment by neobrain** [comment]
Hi, this is neobot, using neobrain's account. I'm keeping an archive of versions of this PR:

2016-02-12T18:50:57Z: https://github.com/citra-emu/citra/compare/f1d1049c4fdf51fd382f9afb6b5cd5f104e71b0e...JamePeng:822a6aed5d062cddd1618febc9245968bddf23bd

**Comment by LittleWhite-tb** [comment]
You have moved ndm_u.cpp in a ndm folder, but not updated the CMakeLists.txt file. CMake is still looking in the service folder for this file.

**Comment by neobrain** [comment]
Hi, this is neobot, using neobrain's account. I'm keeping an archive of versions of this PR:

2016-02-12T19:48:04Z: https://github.com/citra-emu/citra/compare/f1d1049c4fdf51fd382f9afb6b5cd5f104e71b0e...JamePeng:724e0fe42ad6265a40b568559bd40cfafc708246

**Comment by JamePeng** [comment]
@LittleWhite-tb 
No ,that is fault unknown merge,I dont know why it will happen.

NDM change was next request

File: src/core/hle/service/y2r_u.cpp [hunk 18]
```diff
@@ -375,9 +393,9 @@ static void DriverFinalize(Service::Interface* self) {
 
 const Interface::FunctionInfo FunctionTable[] = {
     {0x00010040, SetInputFormat,          "SetInputFormat"},
-    {0x00020000, nullptr,                 "GetInputFormat"},
+    {0x00020000, GetInputFormat,          "GetInputFormat"},
     {0x00030040, SetOutputFormat,         "SetOutputFormat"},
-    {0x00040000, nullptr,                 "GetOutputFormat"},
+    {0x00040000, SetOutputFormat,         "GetOutputFormat"},
```
Comment (linkmauve) [inline src/core/hle/service/y2r_u.cpp]: Typo, should be `GetOutputFormat`.

File: src/core/hle/service/srv.cpp [hunk 19]
```diff
@@ -10,30 +10,34 @@
 
 ////////////////////////////////////////////////////////////////////////////////////////////////////
 // Namespace SRV
-
 namespace SRV {
 
 static Kernel::SharedPtr<Kernel::Event> event_handle;
 
-static void Initialize(Service::Interface* self) {
+ResultCode res = RESULT_SUCCESS;
```
Comment (lioncash) [inline src/core/hle/service/srv.cpp]: This can be replaced by just using RESULT_SUCCESS.raw directly.

File: src/core/hle/service/srv.h [hunk 20]
```diff
@@ -9,6 +9,33 @@
...
+    LaunchFlags7               = 0x117,
+    LaunchFlagsF               = 0x11F,
+    PowerButtonPressed         = 0x202,
+    PowerButtonHeldLong        = 0x203,
+    HomeButtonPressed          = 0x204,
+    HomeButtonReleased         = 0x205,
+    WirelessEnabled            = 0x206,
+    SDcardInserted             = 0x207,
+    GameCartInserted           = 0x208,
+    SDcardRemoved              = 0x209,
```
Comment (lioncash) [inline src/core/hle/service/srv.h]: SDCardRemoved

File: src/core/hle/service/srv.h [hunk 21]
```diff
@@ -9,6 +9,33 @@
...
+    LaunchFlags5               = 0x115,
+    LaunchFlags6               = 0x116,
+    LaunchFlags7               = 0x117,
+    LaunchFlagsF               = 0x11F,
+    PowerButtonPressed         = 0x202,
+    PowerButtonHeldLong        = 0x203,
+    HomeButtonPressed          = 0x204,
+    HomeButtonReleased         = 0x205,
+    WirelessEnabled            = 0x206,
+    SDcardInserted             = 0x207,
```
Comment (lioncash) [inline src/core/hle/service/srv.h]: SDCardInserted

File: src/core/hle/service/srv.h [hunk 22]
```diff
@@ -9,6 +9,33 @@
 
 namespace SRV {
 
+enum class NotificationsID : u32
```
Comment (lioncash) [inline src/core/hle/service/srv.h]: NotificationID

**Comment by neobrain** [comment]
Hi, this is neobot, using neobrain's account. I'm keeping an archive of versions of this PR:

2016-02-12T20:46:43Z: https://github.com/citra-emu/citra/compare/f1d1049c4fdf51fd382f9afb6b5cd5f104e71b0e...JamePeng:dc982addf48235ead1cbefef063cd50395673016

**Comment by lioncash** [comment]
Look, I like the gusto, but can you please stop adding unnecessary things onto your PR alongside code amendments? Your PR title is to implement functions, not modify, but implement. This means no other functions get touched unless the functions being implemented have some relation with another subsystem or module and requires amendments to others in order to function correctly (and the relevant commits document the changes as such).

Doing that makes it increasingly annoying to review changes, because then any reviewer has to look at the entire related codeset to see if it breaks anything when it would be better suited to its own separate PR (I already brought this up twice before) - and to be honest, I honestly don't want to get to the point of reviewing the same slab of code for the umpteenth time because of minor unrelated changes.

**Comment by JamePeng** [comment]
@lioncash Right,this PR only for fix typo now, 
Sorry , but I have a huge knowledge to learn from you! 😀

**Comment by lioncash** [comment]
Can you please amend your recent commit to only fix said typos, then? The additions to AppletUtility are not necessary.

**Comment by JamePeng** [comment]
@lioncash 
Tonight,my master branch was broken three time. One is a buildbot problem ,one is a strange empty merge,the third one revert the 'pull -f' code.
When I have restore the master ,the commits were mess, The mess night make me feel tried.
But I will try to do it well in next step.

**Comment by lioncash** [comment]
What do you mean by 'next step'?

**Comment by JamePeng** [comment]
After this PR

**Comment by lioncash** [comment]
No, please amend your commit.

Use `git commit --amend` to modify your commit after making the changes and then force push with `git push -f`.

**Comment by JamePeng** [comment]
The Applet Utility addition were unnecessary?  I write it because it may help user locate the problem easily!
If unnecessary truly, I will remove it!

**Comment by lioncash** [comment]
> The Applet Utility addition were unnecessary? I write it because it may
help user locate the problem easily!
> If unnecessary truly, I will remove it

Thanks in advance for removing it

**Comment by kitling** [comment]
@JamePeng I think the point is less that it's completely unnecessary and useless, but that it _shouldn't be mixed with these other changes._ If it is still useful, another PR may be in order. But it is not related to "Implementing some functions!" as the title of the PR suggests. Therefore, the changes should be separated into different PRs.

Feel free to submit another PR for the additions to AppletUtility, where they will be properly considered without other 'cruft', such as implementing other functions. Please correct me if I'm wrong, @lioncash .

**Comment by neobrain** [comment]
Hi, this is neobot, using neobrain's account. I'm keeping an archive of versions of this PR:

2016-02-13T08:25:24Z: https://github.com/citra-emu/citra/compare/f1d1049c4fdf51fd382f9afb6b5cd5f104e71b0e...JamePeng:fedeafbe311ac033acc5c846e332c5ce8c31b9ed

**Comment by neobrain** [comment]
Hi, this is neobot, using neobrain's account. I'm keeping an archive of versions of this PR:

2016-02-13T08:54:13Z: https://github.com/citra-emu/citra/compare/f1d1049c4fdf51fd382f9afb6b5cd5f104e71b0e...JamePeng:a48f10a20c7f51f3055ff8feb33c944015627687

**Comment by JamePeng** [comment]
@lioncash Ready for merge?

File: src/core/hle/service/apt/apt_a.cpp [hunk 23]
```diff
@@ -9,25 +9,95 @@ namespace Service {
...
-    {0x000E0080, GlanceParameter,              "GlanceParameter"},
-    {0x000F0100, CancelParameter,              "CancelParameter"},
-    {0x00160040, PreloadLibraryApplet,         "PreloadLibraryApplet"},
-    {0x00180040, PrepareToStartLibraryApplet,  "PrepareToStartLibraryApplet"},
-    {0x001E0084, StartLibraryApplet,           "StartLibraryApplet"},
-    {0x003B0040, nullptr,                      "CancelLibraryApplet?"},
-    {0x00430040, NotifyToWait,                 "NotifyToWait?"},
-    {0x00440000, GetSharedFont,                "GetSharedFont?"},
-    {0x004B00C2, AppletUtility,                "AppletUtility?"},
-    {0x00550040, nullptr,                      "WriteInputToNsState?"},
```
Comment (kitling) [inline src/core/hle/service/apt/apt_a.cpp]: Is it just me, or did this disappear? I'm not finding a corresponding entry added below...

**Pull request closed without merging**